panda/src/navigation/navMeshNode.cxx

```diff
@@ -0,0 +1,101 @@
+/**
+ * PANDA 3D SOFTWARE
+ * Copyright (c) Carnegie Mellon University.  All rights reserved.
+ *
+ * All use of this software is subject to the terms of the revised BSD
+ * license.  You should have received a copy of this license along
+ * with this source code in a file named "LICENSE."
+ *
+ * @file navMeshNode.cxx
+ * @author ashwini
+ * @date 2020-060-21
+ */
+
+
+#include "navMeshNode.h"
+
+TypeHandle NavMeshNode::_type_handle;
+
+NavMeshNode::NavMeshNode(const std::string &name, PT(NavMesh) nav_mesh):
+  PandaNode(name)
+{
+  _nav_mesh = nav_mesh;
+}
+
+NavMeshNode::NavMeshNode(const std::string &name):
+ PandaNode(name) {}
+
+NavMeshNode::~NavMeshNode() {
+  
+}
+
+
+/**
+ * Tells the BamReader how to create objects of type BulletTriangleMeshShape.
+ */
+void NavMeshNode::
+register_with_read_factory() {
+  BamReader::get_factory()->register_factory(get_class_type(), make_from_bam);
+}
+
+/**
+ * Writes the contents of this object to the datagram for shipping out to a
+ * Bam file.
+ */
+void NavMeshNode::
+write_datagram(BamWriter *manager, Datagram &dg) {
+  PandaNode::write_datagram(manager, dg);
+
+  manager->write_pointer(dg, _nav_mesh);
+  // Write NULL pointer to indicate the end of the list.
```

Comment:
The Bullet stuff does this for a list of shapes, but we only have the one NavMesh, so I don't think we need a null value to indicate the end of a list.

Reply:
Removed